src/libraries/Microsoft.PowerFx.Interpreter/Marshal/TypeCoercionProvider.cs

```diff
@@ -238,16 +238,16 @@ internal static bool TryCoerceTo(this FormulaValue value, FormattingInfo formatI
         public static bool TryCoerceToRecord(this RecordValue value, RecordType targetType, out RecordValue result)
         {
             result = null;
-            int n = value.Fields.Count();
-            var recordResult = new NamedValue[n];
-
-            for (int i = 0; i < n; i++)
+            var recordResult = new List<NamedValue>();
+            
+            foreach (var sourceField in value.Fields)
             {
-                var fieldName = value.Fields.ElementAt(i).Name;
+                var fieldName = sourceField.Name;
                 var fieldValue = value.GetField(fieldName);
                 if (!targetType.TryGetFieldType(fieldName, out FormulaType fieldType))
                 {
-                    return false;
+                    recordResult.Add(new NamedValue(fieldName, new BlankValue(IRContext.NotInSource(FormulaType.Blank))));

```

Comment:
>FormulaType.Blank [](http://example.com/codeflow?start=100&length=17)

Should this be the type associated with fieldValue?

Reply:
updated